src/main/java/org/jabref/gui/customentrytypes/CustomEntryTypeDialogViewModel.java

```diff
@@ -126,7 +129,20 @@ public String toString() {
     public void addNewField() {
         Field field = newFieldToAdd.getValue();
         FieldViewModel model = new FieldViewModel(field, true, FieldPriority.IMPORTANT);
-        this.selectedEntryType.getValue().addField(model);
+
+        // create list with all the entry's fields
+        ObservableList<FieldViewModel> entryFields = this.selectedEntryType.getValue().fields();
+
+        // compare every entry field name with the user field name in order to find out if any of them has the same one.
+        boolean fieldExists = entryFields.stream().anyMatch(fieldViewModel -> fieldViewModel.fieldName().getValue().equals(field.getDisplayName()));
+
+        // if the user field name isn't found inside the list, pass it to the entry as a new one. If that is not the case, show warning.
+        if (!fieldExists) {
+            this.selectedEntryType.getValue().addField(model);
+        } else {
+            dialogService.showWarningDialogAndWait(Localization.lang("Duplicate fields"), Localization.lang("Warning: You added field \"%0\" twice. Only one will be kept.", field.getDisplayName()));
+        }
```

Comment:
No code-repeating comments. If the name of the variables are chosen well, then the code is self-explaining and you don't need them.

Reply:
Hello. I deleted the first and last comment and removed as many lines possible. Shall I remove the second comment too? Because I think it is more helpful than the other ones.

Reply:
No, no comment at all, if it's not really necessary and totally misunderstandable otherwise. The best comment is always the code itself, with meaningful variable names and clean code in 98% you don't need any comment at all.
Whoever works with java nowadays knows how to read streams.